python/dgl/transform.py

```diff
@@ -1393,56 +1393,159 @@ def out_subgraph(g, nodes):
+
+    Writeback mappings can be returned for each edge
+    type subgraph. For the edges in the original graph,
+    a writeback mapping is a tensor recording their new
+    ids in the simple graph. If the graph has
+    only one edge type, a single tensor is returned.
+    If the graph has multiple edge types, a dictionary
+    of tensor is returned using canonical edge types
+    as the key.
+
+    Given a :class:`dgl.DGLGraph` object, we return
+    another :class:`dgl.DGLGraph` object representing the
+    simple graph corresponding to it.
 
-    TODO(xiangsx): Don't save writeback_mapping into g, but put it into return value.
 
     Parameters
     ----------
-    g : DGLHeteroGraph
-        The heterogeneous graph
+    g : DGLGraph
+        The input graph.
     return_counts : str, optional
-        If given, the returned graph would have a column with the same name that stores
-        the number of duplicated edges from the original graph.
-    writeback_mapping : str, optional
-        If given, the mapping from the edge IDs of original graph to those of the returned
-        graph would be written into edge feature with this name in the original graph for
-        each edge type.
+        If given, the count of each edge in the original graph
+        will be stored as edge features under the name
+        eturn_counts.
+        (Default: count)
+    writeback_mapping: bool, optional
+        If True, a write back mapping is returned for each edge
+        type subgraph. If False, only the simple graph is returned.
+        (Default: False)
+    copy_ndata: bool, optional
+        If True, the node features of the simple graph are copied
+        from the original graph. If False, the simple
+        graph will not have any node features.
+        (Default: True)
+    copy_edata: bool, optional
+        If True, the edge features of the simple graph are copied
+        from the original graph. If there exists duplicate edges between
+        two nodes (u, v), the feature of the edge is randomly selected
+        from one of the duplicate edges.
+        If False, the simple graph will not have any edge features.
+        (Default: False)
 
     Returns
     -------
-    DGLHeteroGraph
-        The new heterogeneous simple graph.
+    DGLGraph
+        A simple graph.
+    tensor or dict of tensor
+        If writeback_mapping is True, the writeback
+        mapping is returned. If the graph has only
+        one edge type, a tensor is returned. If the
+        graph has multiple edge types, a dictionary
+        of tensor is return.
+
+    If ``copy_ndata`` is ``True``, same tensors will be used for
+    the features of the original graph and the to_simpled graph. As a result, users
+    should avoid performing in-place operations on the features of the to_simpled
+    graph, which will corrupt the features of the original graph as well. For
+    concrete examples, refer to the ``Examples`` section below.
```

Comment:
Try to avoid passive tense. Also, I will not explicitly **suggest** anything while just describing what will happen.
```
If ``copy_ndata=True``, it shallow-copies node feature tensors to the result graph.
In-place operations on the features of the result graph may change the features
of the original graph too (depending on the shallow-copy semantics of the backend
framework in-use).
```
@mufeili  please take a look too. This paragraph will appear in many API doc.

I also suggest putting it before the `Parameters` section and surround it with `..note::` block.

Reply:
I think we can refine the docstring later. How to describe copy_ndata is not well defined right now.